R/workhorse.R

```diff
@@ -75,22 +75,24 @@ make_request <- function(x, ...){
 }
 
 process_request <- function(res,
-                            content = TRUE,
-                            expected = "application/json; charset=UTF-8") {
-  httr::stop_for_status(res)
-  if (content == TRUE) {
-    actual <- res$headers$`content-type`
-    if (actual != expected) {
-      spf(
-        paste0(
-          "Expected content-type:\n%s",
-          "\n",
-          "Actual content-type:\n%s"
-        ),
-        expected,
-        actual
-      )
-    }
-    jsonlite::fromJSON(httr::content(res, "text"), simplifyVector = FALSE)
+                            expected = "application/json; charset=UTF-8",
+                            internet = TRUE) {
```

Comment:
If the problem is test input, i.e. it's not a full-fledged httr response object, here's an option:

Come up with a call that creates such an object with the properties you need. Use it to capture a response. Comment it out. Save this as a "reference object" and use `testthat::expect_equal_to_reference()`. Or just generally load as needed, in order to test response processors.